omigami/plotting.py

```diff
@@ -0,0 +1,153 @@
+
+            mol_render_list.append(molecule)
+
+        image = Draw.MolsToGridImage(mol_render_list, subImgSize=molecule_image_size,
+                                     legends=spectra_matches.compound_name.tolist(),
+                                     highlightBondLists=highlight_bonds)
+        return image
+
+    @staticmethod
+    def plot_classyfire_result(smiles_list: List[str], color="g"):
+        """TODO: Ask joe what those classifiers actually classfy"""
+        class_stats = dict()
+        for smiles in smiles_list:
+            try:
+                classyfire_result = requests.get(CLASSYFIRE_URL + smiles).json()
+                class_assignment = classyfire_result['class']['name']
+
+                if class_assignment in class_stats.keys():
+                    class_stats[class_assignment] += 1
+                elif class_assignment not in class_stats.keys():
+                    class_stats[class_assignment] = 1
+            except:
+                pass
+
+        return plt.barh(list(class_stats.keys()), class_stats.values(), color=color)
+
+    @staticmethod
+    def plot_NPclassifier_result(smiles_list: List[str], color="g"):
+        class_stats = dict()
+        class_stats['Cannot_Assign'] = 0
+        for smiles in smiles_list:
+
+            try:
+                NPclassifier_result = requests.get(NPCLASSIFER_URL + smiles).json()
+                class_assignment = NPclassifier_result['superclass_results'][0]
+                if class_assignment in class_stats.keys():
+                    class_stats[class_assignment] += 1
+                elif class_assignment not in class_stats.keys():
+                    class_stats[class_assignment] = 1
+
+            except:
+                class_stats['Cannot_Assign'] += 1
+        return plt.barh(list(class_stats.keys()), class_stats.values(), color=color)
+
+    @staticmethod
+    def _validate_data(spectra_matches: pd.DataFrame, representation: str = 'smiles'):
+        if representation not in ["smiles", "inchi"]:
+            raise ValueError(
+                f"Got unexpected representation string. Needs to be either 'smiles' or 'inchi' got {representation}"
+            )
+
+        if not isinstance(spectra_matches, pd.DataFrame):
+            raise ValueError(
+                f"Matches need to be a Pandas DataFrame got {type(spectra_matches)}"
+            )
+
+        if "compound_name" not in spectra_matches.columns:
+            raise MandatoryColumnMissingError("The provided DataFrame must contain a column named compound_name")
+
+    @staticmethod
+    def _clean_matches(spectra_matches: pd.DataFrame, representation: str) -> pd.DataFrame:
+        spectra_matches = spectra_matches.drop_duplicates('compound_name')
+        spectra_matches = spectra_matches.drop_duplicates(representation)
+        spectra_matches = spectra_matches[spectra_matches[representation] != ""]
+        spectra_matches = spectra_matches.dropna()
```

Comment:
I think here you should specify that you mean drop if it's na in the representation column, and not all na in general